oat-sa/extension-tao-test-runner-plugins
# Feature/nec 40/Improve the answerCache plugin

Related to: https://oat-sa.atlassian.net/browse/NEC-40

Rework the answer cache plugin:
- add an option to take care of all item attempts, instead of the first one only
- make it resilient to navigation issues: clear the cache only after having successfully reached the next step in the navigation (resilient to connectivity issues as well as session timeout or disconnections)

How to test it:
- install the extension `taoTestRunnerPlugins`
- enable the `answerCache` plugin (in the file `config/taoTests/test_runner_plugin_registry.conf.php`)
- set the plugin option `allAttempts` to true (in the file `config/taoQtiTest/testRunner.conf.php`)
- take a test
    - answer to an item
    - refresh the page
    - the answer should be reloaded
- take a test
    - answer to an item
    - stop the server
    - go to next item, the test should fail
    - restart the server
    - refresh the page
    - the answer should be reloaded
- take a test
    - answer to an item
    - kill the session (you may open another tab on the same session, using the home link, and logout the user)
    - reconnect and resume the test
    - the answer should be reloaded

Unit tests should work:
```
cd tao/views/build
npx grunt connect:dev taotestrunnerpluginstest
```

Or from the browser: `http://<YOUR_HOST>/taoTestRunnerPlugins/views/js/test/runner/plugins/content/answerCache/test.html`

Impacted test: `taoTestRunnerPlugins/views/js/test/runner/plugins/content/answerCache/test.html`
